### ISSUE TITLE
Qu'est-ce qu'un webhook ?

### ISSUE THREAD

**Issue by christianfelicite** [TheFeloDevTeam/FeloFamilySite#111]

**Comment by christianfelicite** [comment]
https://docs.github.com/en/developers/webhooks-and-events/about-webhooks

**Comment by christianfelicite** [comment]
https://developer.atlassian.com/server/jira/platform/webhooks/

A webhook is a user-defined callback over HTTP. You can use Jira webhooks to notify your app or web application when certain events occur in Jira. For example, you might want to alert your remote application when an issue is updated or when sprint is started. Using a webhook to do this means that your remote application doesn't have to periodically poll Jira (via the REST APIs) to determine whether changes have occurred.

**Comment by christianfelicite** [comment]
Overview

A webhook in Jira is defined by the following information that you need to provide when registering (that is, creating) a new webhook:

    A name for the webhook created (for example, "Webhook for my remote app").
    The URL where the callback should be sent.
    The scope of the webhook, either "all issues" or a limited set of issues specified by JQL ("project = TEST and fixVersion = future").
    The events to post to the URL, either "all events" or a specific set of events.

A simple webhook might look like this:

Name: "My simple webhook"
URL: www.myremoteapp.com/webhookreceiver


Scope: all issues
Events: all issue events

A more complex webhook might look like this:

Name: "A slightly more advanced webhook"
URL: www.myremoteapp.com/webhookreceiver

Scope: Project = JRA AND fixVersion IN ("6.4", "7.0")
Events: Issue Updated, Issue Created

**Comment by christianfelicite** [comment]
There are two ways your apps can communicate with each other to share information: polling and webhooks. As one of our customer champion's friends has explained it: Polling is like knocking on your friend’s door and asking if they have any sugar. Webhooks are like someone tossing a bag of sugar at your house whenever they buy some.

Webhooks are automated messages sent from apps when something happens. They have a message—or payload—and are sent to a unique URL—essentially the app's phone number or address.

They're much like SMS notifications. Say your bank sends you an SMS when you make a new purchase. You already told the bank your phone number, so they knew where to send the message. They type out "You just spent $10 at NewStore" and send it to your phone number +1-234-567-8900. Something happened at your bank, and you got a message about it. All is well.

Webhooks work the same way.

https://zapier.com/blog/what-are-webhooks/

**Comment by christianfelicite** [comment]
When an event happens on the trigger application, it serializes data about that event and sends it to a webhook URL from the action application—the one you want to do something based on the data from the first application. The action application can then send a callback message, often with an HTTP status code like 302 to let the trigger application know if the data was received successfully or 404 if not.

**Comment by christianfelicite** [comment]
Webhooks are similar to APIs—but simpler. An API is a full language for an app with functions or calls to add, edit, and retrieve data. The difference is, with an API, you have to do the work yourself. If you build an application that connects to another with an API, your application will need to have ways to ask the other app for new data when it needs it. Webhooks, on the other hand, are for one specific part of an app, and they're automated. You might have a webhook just for new contacts—and whenever a new contact is added, the application will push the data to the other application's webhooks URL automatically. It's a simple, one-to-one connection that runs automatically.

**Comment by christianfelicite** [comment]
https://blog.cloud-elements.com/webhooks-vs-polling-youre-better-than-this

![1.png](https://user-images.githubusercontent.com/24352234/87853615-deeb3080-c90b-11ea-82a7-8086f4fb1eb9.png)

**Comment by christianfelicite** [comment]
https://fr.mailjet.com/blog/news/webhook/

**Comment by christianfelicite** [comment]
https://www.getvero.com/resources/webhooks/

![webhooks-3.png](https://user-images.githubusercontent.com/24352234/87907635-44066980-ca65-11ea-8b2c-9257cb908a5f.png)

![webhooks-1-1-1.png](https://user-images.githubusercontent.com/24352234/87907647-47015a00-ca65-11ea-885b-c99456cd7071.png)

**Comment by christianfelicite** [comment]
https://www.getvero.com/resources/webhooks/

Webhooks vs. APIs

You’ll often hear APIs and webhooks mentioned together. And while they’re similar in what they can help you achieve – they’re not the same thing.

As mentioned earlier, webhooks are just one of the ways that different applications use to communicate with each other, and another is through an application programming interface (API).

Their uses cases are very similar but the prime difference between API and webhooks?

It’s in how they receive data.
Pulling vs Pushing

With an API, you get data through a process known as “polling.” This is when your application periodically makes a request to an API server to check for new data.

    A webhook, on the other hand, allows the provider to send (i.e “push”) data to your application as soon as an event occurs. This is why webhooks are sometimes referred to as “reverse APIs.”

APIs need to pull data from a server periodically to stay up to date, but with webhooks, the server can push this data over to you the instant something happens.

stripe webhooks

To use a real world analogy, APIs would be likened to you repeatedly calling a retailer to ask if they’ve stocked up on a brand of shoes you like. Webhooks, would then be like asking the retailer to call you whenever they have the shoes in stock, which frees up time on both sides.

Webhooks are less resource-intensive because they save you time on constantly polling (checking) for new data.

Here’s a diagram that shows how APIs and webhooks handle new data:

webhooks vs. apis

Even though only a single event occurred at 1 pm, you’d have to constantly poll your provider for new data with an API. With webhooks, you just get notified once at 1pm. It’s vastly more productive.

    So the question is, if webhooks are easier to set up, less resource-intensive, and faster than APIs, why use APIs at all?

Well, APIs are still popular for a number of reasons:

    Not every application supports webhook integrations.
    Sometimes you only want to know about the end result, rather than every event (i.e. every permutation) that’s changed an object.
    Webhooks can only notify you about an event, so if you want to make a change based on new information, you’ll need an API.
    A webhook payload may not contain all the data you need about an event.

So APIs are still really useful, which is why a lot of applications support both APIs and webhooks.

Bottom line is, if your goal is to transfer data between two services, webhooks are the way to go.

**Comment by christianfelicite** [comment]
https://codeburst.io/what-are-webhooks-b04ec2bf9ca2

**Comment by christianfelicite** [comment]
#146
